modules/wc-product-editor/assets/js/e-wc-product-editor.js

```diff
@@ -0,0 +1,63 @@
+import { WooHeaderItem } from '@woocommerce/admin-layout';
+import { useEntityId } from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import { registerPlugin } from '@wordpress/plugins';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+function EditWithElementorButton() {
+	const productId = useEntityId( 'postType', 'product' );
+	const { saveEntityRecord } = useDispatch( 'core' );
+
+	const postStatus = useSelect( ( select ) => {
+		return select( 'core' ).getEditedEntityRecord( 'postType', 'product', productId )?.status;
+	}, [ productId ] );
+
+	const handleClick = () => {
+		if ( 'auto-draft' === postStatus ) {
+			saveEntityRecord( 'postType', 'product', {
```

Comment:
This should work fine, but one alternative to this it to use the [useProductManager](https://github.com/woocommerce/woocommerce/blob/8c523190d86aa9d628e146b6dca22f8fcaa4097a/packages/js/product-editor/src/hooks/use-product-manager/use-product-manager.ts#L47) hook exported from `@woocommerce/product-editor` as `__experimentalUseProductManager`.

This hook has a number of states and methods you can use:

```js
return {
	isValidating,
	isDirty,
	isSaving,
	isPublishing: isSaving,
	isTrashing,
	save,
	publish,
	trash,
	copyToDraft,
};
```

Reply:
Hey @joshuatf, I think I prefer not to bring yet another package in just for the save, but thanks, it was helpful.

PS: Again got `ERESOLVE unable to resolve dependency tree` when trying to install `@woocommerce/product-editor`

Still keen to hear your feedback if you have further push back or thoughts :)

Reply:
That should be fine!  It does the same thing for now.  The one benefit to `useProductManager` is that you'd get progressive updates to the hook functions as we go along, but it may not be necessary in your case.

Also, we have no plans to move away from the entity data store so this should continue to work without issue.